### PR TITLE
Boos 319/feature/postgres

### PR DESCRIPTION
**Pull request by koutsilisk**
In this PR we introduce:
- Posgresql database configuration 
- Some more docker-configuration configuration

Fixes:
- Activate users when population from manage.py
- Email service working from within docker

### CODE REVIEW

**Comment by koutsilisk** [comment]
MERGE AFTER #122

File: .gitignore [hunk 1]
```diff
@@ -27,3 +27,4 @@ frontend/docker-compose.yml
 
 .vscode
 .idea
+env_files
```
Comment (Chichilele) [inline .gitignore:30]: empty line

File: README.md [hunk 2]
```diff
@@ -2,20 +2,32 @@
...
-
 ## Start app
 
-Simply `docker-compose up` to bring the app up and go to http://localhost:3000/admin.
+1. Write the following env files in `env_files`. Examples can be found in folder `env_example`:
+
+   - `api.env`: mainly flask related variables
+   - `postgres.env`: database variables
+
+2. Simply `docker-compose --env-file ./env_files/api.env up` to bring the app up.
```
Comment (Chichilele) [inline README.md:12]: Wouldn't it be better to have the files pointed at in the docker-compose file?

File: env_examples/postgres.env [hunk 3]
```diff
@@ -0,0 +1,5 @@
+POSTGRES_USER=postgres
+POSTGRES_PASSWORD=postgres
+FLASK_DATABASE_NAME=flask_api_db
+FLASK_DATABASE_USER=flask_api_user
+FLASK_DATABASE_PASSWORD=flask_api_password
```
Comment (Chichilele) [inline env_examples/postgres.env:5]: empty line

File: env_examples/api.env [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+FLASK_DATABASE_USER=flask_api_user
+FLASK_DATABASE_PASSWORD=flask_api_password
+# FLASK_DATABASE_PORT=...
+
+### Email VARIABLES ###
+# FLASK_MAIL_USERNAME=...
+# FLASK_MAIL_PASSWORD=...
+# FLASK_MAIL_DEFAULT_SENDER_NAME=...
+# FLASK_MAIL_DEFAULT_SENDER_EMAIL=...
+FLASK_MAIL_PORT=587
```
Comment (Chichilele) [inline env_examples/api.env:19]: empty line

File: docker-compose.yml [hunk 5]
```diff
@@ -10,13 +10,34 @@ services:
...
+    ports: 
+      - 5432:5432
+    env_file:
+      - ./env_files/postgres.env
+    volumes:
+      - postgres:/var/lib/postgresql/data/
+    restart: always
+
+volumes:
+  postgres:
```
Comment (Chichilele) [inline docker-compose.yml:43]: empty line

**Review by Chichilele — CHANGES_REQUESTED** [review]
just add empty lines at the end of each file

File: README.md (hunk 2 above)
Comment (koutsilisk) [reply README.md:12]: we have them but in order to expose the mail port and not to have it as a static port i added this, and in docker-compose we have `- ${FLASK_MAIL_PORT}:${FLASK_MAIL_PORT}`. That's the reason. I could make it static ( example 587:587) and add a comment that this is the mail port

Comment (Chichilele) [reply README.md:12]: that's not exactly what I meant. You can point to a config file directly form the docker-compose file I think? have a look at our config in boost.

File: docker-compose.yml [hunk 6]
```diff
@@ -10,13 +10,35 @@ services:
...
+
```
Comment (Chichilele) [inline docker-compose.yml:44]: the last line should be enmpty

**Review by Chichilele — APPROVED** [review]

File: README.md (hunk 2 above)
Comment (koutsilisk) [reply README.md:12]: in order not to use the --env-file we should have a .env file at the same folder as the docker-compose file as mentioned here: https://docs.docker.com/compose/environment-variables/#the-env-file, the --env-file helps to save the configuration in any folder you want with any name you want

File: Makefile [hunk 7]
```diff
@@ -0,0 +1,11 @@
...
+	@docker-compose --env-file ./env_files/api.env up
+
+stop:
+	@docker-compose --env-file ./env_files/api.env down
+
+build:
+	@docker-compose --env-file ./env_files/api.env build
+
+logs: 
+	@docker-compose --env-file ./env_files/api.env logs -f
```
Comment (Chichilele) [inline Makefile:11]: empty line

**Review by Chichilele — APPROVED** [review]

**Pull request merged**